src/OpenApi/src/Services/Schemas/OpenApiSchemaService.cs

```diff
@@ -88,9 +88,10 @@ internal sealed class OpenApiSchemaService(
             {
                 schema = new JsonObject();
             }
-            schema.ApplyPrimitiveTypesAndFormats(context);
-            schema.ApplySchemaReferenceId(context);
-            schema.ApplyPolymorphismOptions(context);
+            var createSchemaReferenceId = optionsMonitor.Get(documentName).CreateSchemaReferenceId;
```

Comment:
Use `_openApiOptions`?

Reply:
`_openApiOptions` is an instance field so we can't use it in the initializer for `_confguration`. I could switch to standard constructors instead of primary constructors and initialize everything there but.... 😅 🤪